src/core_functions/scalar/map/map.cpp

```diff
@@ -32,12 +32,12 @@ static void AlignVectorToReference(const Vector &original, const Vector &referen
 
 	Vector expanded_const(ListType::GetChildType(original.GetType()), new_length);
 
-	auto expansion_factor = new_length / original_length;
-	if (expansion_factor != tuple_count) {
+	if (new_length != tuple_count * original_length) {
 		throw InvalidInputException("Error in MAP creation: key list and value list do not align. i.e. different "
 		                            "size or incompatible structure");
 	}
-	CreateExpandedVector(original, expanded_const, expansion_factor);
+	if (original_length)
```

Comment:
Do we need this `if`?

Reply:
it can be potentially turned into a `D_ASSERT(original_length)`, but I have no idea if that's actually the case.

Idea is that we should avoid division by 0, but I have no actual idea of the underlying code to understand what is the best way to avoid that, it might be that I am too conservative.

Reply:
Makes sense. Let's rewrite it to the following, then. Besides this, it looks good to me.
```cpp
auto expansion_factor = original_length ? new_length / original_length : 0;
CreateExpandedVector(original, expanded_const, expansion_factor);
```